File: app/src/prod/java/com.elegion.android.app/repository/OkHttpProvider.java
package com.elegion.android.app.repository;

import android.support.annotation.NonNull;

import com.elegion.android.BuildConfig;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

/**
 * @author Daniel Serdyukov
 */
public final class OkHttpProvider {

    private OkHttpProvider() {
        throw new IllegalStateException("Final class can not be instantiated");
    }

    @NonNull
    public static OkHttpClient provideClient() {
        final OkHttpClient.Builder builder = new OkHttpClient.Builder();
        if (BuildConfig.DEBUG) {
            builder.addInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY));
        }
        return new OkHttpClient.Builder().build();
    }

}

File: app/src/main/kotlin/com/elegion/android/bankex/data/provider/OkHttpProvider.kt
package com.elegion.android.bankex.data.provider

import com.elegion.android.bankex.BuildConfig

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

object OkHttpProvider {

    @JvmStatic
    val okHttpClient: OkHttpClient by lazy {
        provideClient()
    }

    @JvmStatic
    fun provideClient(): OkHttpClient {
        val builder = OkHttpClient.Builder()
        // logs
        if (BuildConfig.DEBUG) {
            builder.addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
        }
        return builder.build()
    }
}
